great_expectations/data_context/types/base.py

```diff
@@ -1063,6 +1063,57 @@ def make_notebooks_config(self, data, **kwargs):
         return NotebooksConfig(**data)
 
 
+class ConcurrencyConfig(DictDot):
+    """WARNING: This class is experimental."""
+
+    def __init__(self, enabled: Optional[bool] = False):
+        """Initialize a concurrency configuration to control multithreaded execution.
+
+        Args:
+            enabled: Whether or not multithreading is enabled.
+        """
+        self._enabled = enabled
+
+    @property
+    def enabled(self):
+        """Whether or not multithreading is enabled."""
+        return self._enabled
+
+    @property
+    def max_database_query_concurrency(self) -> int:
+        """Max number of concurrent database queries to execute with mulithreading."""
+        # BigQuery has a limit of 100 for "Concurrent rate limit for interactive queries" as described at
```

Comment:
follow on PR to user configure 100

Reply:
just a reminder that when we do this, we should update the marshmallow schema as well.